### PR TITLE
AuthenticationStrengthv1

### PR DESCRIPTION
**Pull request by InbarckMS**
Moving Auth Strength to v1 and depreacting functions from beta

### CODE REVIEW

File: api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md [hunk 1]
```diff
@@ -7,11 +7,15 @@ ms.prod: "identity-and-sign-in"
...
 ---
 
-# authenticationStrengthPolicy: findByMethodMode
+# authenticationStrengthPolicy: findByMethodMode (deprecated))
 Namespace: microsoft.graph
 
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
+[!CAUTION] 
+  > The `findByMethodMode` function is only available in `/beta` and will not become available in `/v1.0`. To get a list of  [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) filtered to only include policies that include the authentication method mode specified in the request, use XXX instead
```
Comment (jbzdarkid) [inline api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:16]:
```suggestion
  > The `findByMethodMode` function is only available in `/beta` and will not become available in `/v1.0`. To get a list of [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) filtered to only include policies that include the authentication method mode specified in the request, use XXX instead
```

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+|[updateAllowedCombinations](../api/authenticationstrengthpolicy-updateallowedcombinations.md)|[updateAllowedCombinationsResult](../resources/updateallowedcombinationsresult.md)|Update the allowed [authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) for a given [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md).|
+|[List combinationConfigurations](../api/authenticationstrengthpolicy-list-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) collection|Get the authenticationCombinationConfiguration resources from the combinationConfigurations navigation property.|
+|[Create authenticationCombinationConfiguration](../api/authenticationstrengthpolicy-post-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md)|Create a new authenticationCombinationConfiguration object.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationstrengthpolicy.md:29]: Not here. They should be in the authenticationCombinationConfiguration.md file.
```suggestion
```

File: api-reference/v1.0/resources/authenticationstrengthusage.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationstrengthusage.md:1]: This object isn't in the v1.0 schema. Please verify.
Once fixed, reassign the doc to yourself.

File: api-reference/v1.0/api/authenticationstrengthpolicy-updateallowedcombinations.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
+---
+title: "authenticationStrengthPolicy: updateAllowedCombinations"
+description: "Update the allowedCombinations collection for a given authentication strength policy object."
+author: "mmcla; inbarckMS"
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/authenticationstrengthpolicy-updateallowedcombinations.md:4]: Only one author.

File: api-reference/v1.0/resources/authenticationcombinationconfiguration.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ms.localizationpriority: medium
+ms.prod: "identity-and-sign-in"
+doc_type: resourcePageType
+---
+
+# authenticationCombinationConfiguration resource type
+
+Namespace: microsoft.graph
+
+An authentication combination configuration sets restrictions on specific types, modes, versions, etc. of an authentication method that is tied to specific auth method combinations used in an [authentication strength](authenticationstrengths-overview.md).
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationcombinationconfiguration.md:14]:
```suggestion
An authentication combination configuration sets restrictions on specific types, modes, and versions of an authentication method that is tied to specific auth method combinations used in an [authentication strength](authenticationstrengths-overview.md).
```

File: api-reference/v1.0/resources/authenticationcombinationconfiguration.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+|[Update](../api/authenticationcombinationconfiguration-update.md)|[fido2combinationconfiguration](../resources/fido2combinationconfiguration.md)|Update the properties of a [fido2combinationconfiguration](../resources/fido2combinationconfiguration.md) object.|
+|[Delete](../api/authenticationstrengthpolicy-delete-combinationconfigurations.md)|None|Deletes an [fido2combinationconfiguration](../resources/fido2combinationconfiguration.md) object.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|appliesToCombinations|authenticationMethodModes collection|Which authentication method combinations this configuration applies to. Must be an **allowedCombinations** object that's defined for the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md). The only possible value for fido2combinationConfigurations is `"fido2"`.|
+|id|String|A unique system-generated identifier. Inherited from [entity](../resources/entity.md).|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationcombinationconfiguration.md:33]:
```suggestion
|id|String|A unique system-generated identifier.|
```

File: api-reference/v1.0/resources/authenticationcombinationconfiguration.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+## Relationships
+None.
+
+## JSON representation
+The following is a JSON representation of the resource.
+<!-- {
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationCombinationConfiguration",
+  "baseType": "microsoft.graph.entity",
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationcombinationconfiguration.md:44]: For same reason indicated above
```suggestion
```

File: api-reference/v1.0/resources/authenticationmethodmodedetail.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+# authenticationMethodModeDetail resource type
+
+Namespace: microsoft.graph
+
+The details of the **authenticationMethodModes** objects that can be defined for the **allowedCombinations** property of the [authenticationstrengthpolicy](../resources/authenticationStrengthPolicy.md).
+
+For more information on authentication methods, see the [authentication methods overview](../resources/authenticationmethods-overview.md)
+
+Inherits from [entity](../resources/entity.md).
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationmethodmodedetail.md:18]: Not anymore

File: api-reference/v1.0/resources/authenticationmethodmodedetail.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+|[List authenticationMethodModeDetails](../api/authenticationstrengthroot-list-authenticationmethodmodes.md)|[authenticationMethodModeDetail](../resources/authenticationmethodmodedetail.md) collection|Get a list of the [authenticationMethodModeDetail](../resources/authenticationmethodmodedetail.md) objects and their properties.|
+|[Get authenticationMethodModeDetail](../api/authenticationmethodmodedetail-get.md)|[authenticationMethodModeDetail](../resources/authenticationmethodmodedetail.md)|Read the properties and relationships of an [authenticationMethodModeDetail](../resources/authenticationmethodmodedetail.md) object.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|authenticationMethod|baseAuthenticationMethod|The authentication method that this mode modifies. The possible values are: `password`, `voice`, `hardwareOath`, `softwareOath`, `sms`, `fido2`, `windowsHelloForBusiness`, `microsoftAuthenticator`, `temporaryAccessPass`, `email`, `x509Certificate`, `federation`, `unknownFutureValue`.|
+|displayName|String|The display name of this mode|
+|id|String|The system-generated identifier for this mode. Inherited from [entity](../resources/entity.md).|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationmethodmodedetail.md:31]: Same reason as above
```suggestion
|id|String|The system-generated identifier for this mode. |
```

File: api-reference/v1.0/resources/authenticationmethodmodedetail.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+## Relationships
+None.
+
+## JSON representation
+The following is a JSON representation of the resource.
+<!-- {
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationMethodModeDetail",
+  "baseType": "microsoft.graph.entity",
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/authenticationmethodmodedetail.md:42]: Same reason as above
```suggestion
```

File: api-reference/v1.0/api/authenticationstrengthroot-list-authenticationmethodmodes.md [hunk 11]
```diff
@@ -0,0 +1,240 @@
+---
+title: "List authenticationMethodModes"
+description: "Get a list of all supported authentication methods, or all supported authentication method combinations."
+author: "mmcla;inbarckMS"
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/authenticationstrengthroot-list-authenticationmethodmodes.md:4]: Only One

File: api-reference/v1.0/resources/conditionalaccessgrantcontrols.md [hunk 12]
```diff
@@ -33,7 +33,10 @@ Consider the following when you use the `passwordChange` control:
 
 ## Relationships
 
-None.
+|Relationship|Type|Description|
+|:---|:---|:---|
+| authenticationStrength | [authenticationStrengthPolicy](authenticationstrengthpolicy.md) | The authentication strength required by the conditional access policy. Optional.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/conditionalaccessgrantcontrols.md:38]: This change isn't in the v1.0 schema PR. Please verify.

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @InbarckMS for your work on this PR.

I've left suggestions inline plus the following:
- Question: Are updateAllowedCombinations action and findByMethodMode function getting deprecated in beta or is it that they won't be promoted to v1.0? 
   - If it's the latter, remove the "(deprecated)" statement from both the TOC and the titles.
   - If the former, also ensure that your schema PR includes the deprecation annotations. That will be required for approval of the docs PR.
- Add the **authenticationStrengthPolicies** relationship to [policyRoot](https://learn.microsoft.com/en-us/graph/api/resources/policyroot?view=graph-rest-1.0).
- GLOBAL: The enums like authenticationMethodModes, authenticationStrengthPolicyType, and authenticationStrengthRequirements haven't been defined in the enums.md file . This is the cause for many (if not all) of the API doctor errors in the v1.0 files.
- GLOBAL: I mentioned a while back that we need to start calling out the RBAC roles for delegated access READ vs WRITE. That way, customers don't have to use Global Admin, dig around, or report 403s. Let me know once you have the info available and I can guide you on this.
   - Let's do that for GA in this PR. You can raise a separate PR for beta.
- GLOBAL: Delete all the `# [HTTP](#tab/http)` in examples. These will be added by the snippets generator.
- Couldn't find neither the **usage** function nor the **authenticationStrengthUsage** complex type in the v1.0 schema PR. Please verify.
- Couldn't find the **authenticationStrength** relationship of conditionalAccessGrantControls in the schema either. Please verify.
- GLOBAL: I see some docs are still assigned to Michael. Please reassign them to yourself. Even the beta docs.
  - Also, only 1 alias is allowed in the "author" field.
- GLOBAL: None of the resources inherited from entity in v1.0. Please make sure you remove any "Inherits from entity/Inherited from entity/"baseType": "microsoft.graph.entity"" entries.
- GLOBAL: I see you moved all CA-related URLs from `/identity/conditionalAccess/authenticationStrengths/` (plural Strengths) to `/identity/conditionalAccess/authenticationStrengths/` (singular Strength). However, there are many left over instances even in beta.
   - Please do a search and replace to replace all these instances in both v1.0 and beta.
   - For response objects, the @odata.context also contains the plural form. Make sure to change to the singular form `...#identity/conditionalAccess/authenticationStrength/`.
   - This isn't a breaking change because you're not blocking the API side. But let's document the endpoint that you want customers to use going forward.
- Add the changelog please. Remember to download it from your schema PR in time. The artifacts expire after 15 days of a successful build and if the schema PR is merged and 15 days pass before you get the files, you will need to create the changelog manually.

File: api-reference/beta/api/authenticationstrengthpolicy-updateallowedcombinations.md [hunk 13]
```diff
@@ -7,11 +7,12 @@ ms.prod: "identity-and-sign-in"
 doc_type: apiPageType
 ---
 
-# authenticationStrengthPolicy: updateAllowedCombinations
+# authenticationStrengthPolicy: updateAllowedCombinations
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/beta/api/authenticationstrengthpolicy-updateallowedcombinations.md:10]: We ended up not depreacting updateAllowedCombinations, only findByMethod.

File: api-reference/v1.0/api/authenticationstrengthpolicy-usage.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
+---
+title: "authenticationStrengthPolicy: usage"
+description: "The `usage` function allows the caller to see which Conditional Access policies reference a specified authentication strength policy"
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthpolicy-usage.md:3]: Updated

File: api-reference/v1.0/api/authenticationstrengthpolicy-usage.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+author: "inbarckMS"
+ms.localizationpriority: medium
+ms.prod: "identity-and-sign-in"
+doc_type: apiPageType
+---
+
+# authenticationStrengthPolicy: usage
+Namespace: microsoft.graph
+
+The `usage` function allows the caller to see which Conditional Access policies reference a specified authentication strength policy. The policies are returned in two collections, one containing Conditional Access policies that require an MFA claim and the other containing Conditional Access policies that do not require such a claim. Policies in the former category are restricted in what kinds of changes may be made to them to prevent undermining the MFA requirement of those policies.
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthpolicy-usage.md:13]: Updated

File: api-reference/v1.0/api/authenticationstrengthroot-list-policies.md [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+|Permission type|Permissions (from least to most privileged)|
+|:---|:---|
+|Delegated (work or school account)|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+|Delegated (personal Microsoft account)|Not supported.|
+|Application|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthroot-list-policies.md:23]: These are the RBAC:

-	Conditional Access administrator (R/W)
-	Security Administrator (R/W)
-	Security Reader (Read only)

Can you share an example I can modify for auth strength?

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+|[Get authenticationStrengthPolicy](../api/authenticationstrengthpolicy-get.md)|[authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md)|Read the properties and relationships of an [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object.|
+|[Update authenticationStrengthPolicy](../api/authenticationstrengthpolicy-update.md)|[authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md)|Update the properties of a custom [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object. You can't update a built-in **authenticationStrengthPolicy** object. |
+|[Delete authenticationStrengthPolicy](../api/authenticationstrengthroot-delete-policies.md)|None|Delete a custom [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object. You can't delete a built-in **authenticationStrengthPolicy** object.|
+|[List usage](../api/authenticationstrengthpolicy-usage.md)|[authenticationStrengthUsage](../resources/authenticationstrengthusage.md)|Find all [conditionalAccessPolicies](../resources/conditionalaccesspolicy.md) that reference an authentication strength.|
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthpolicy.md:26]: We are going to add it to the PR, I'm keeping this discussion open until we do.

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md (hunk 2 above)
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthpolicy.md:29]: I see them already in authenticationCombinationConfiguration, so I'm going to only remove them from here.

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+|[List combinationConfigurations](../api/authenticationstrengthpolicy-list-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) collection|Get the authenticationCombinationConfiguration resources from the combinationConfigurations navigation property.|
+|[Create authenticationCombinationConfiguration](../api/authenticationstrengthpolicy-post-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md)|Create a new authenticationCombinationConfiguration object.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|allowedCombinations|authenticationMethodModes collection|A collection of authentication method modes that are required be used to satify this authentication strength.|
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthpolicy.md:30]: Updated. Will check API doctor again after the next commit.

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+|createdDateTime|DateTimeOffset|The datetime when this policy was created.|
+|description|String|The human-readable description of this policy.|
+|displayName|String|The human-readable display name of this policy. <br><br>Supports `$filter` (`eq`, `ne`, `not` , and `in`). |
+|id|String|The system-generated identifier for this mode. Inherited from [entity](../resources/entity.md).|
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthpolicy.md:34]: Updated.

File: api-reference/v1.0/resources/authenticationstrengthroot.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+---
+
+# authenticationStrengthRoot resource type
+
+Namespace: microsoft.graph
+
+
+The **authenticationStrengthRoot** resource is the entry point for the authentication strengths object model.
+
+Inherits from [entity](../resources/entity.md).
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthroot.md:17]: Updated

File: api-reference/v1.0/resources/authenticationstrengthroot.md [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+
+## Methods
+
+None.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|authenticationCombinations|authenticationMethodModes collection|A collection of all valid authentication method combinations in the system.|
+|id|String|A system-generated identifier. Inherited from [entity](../resources/entity.md).|
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthroot.md:27]: Updated

File: api-reference/v1.0/resources/authenticationstrengthroot.md [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+|authenticationMethodModes|[authenticationMethodModeDetail](../resources/authenticationmethodmodedetail.md) collection|Names and descriptions of all valid authentication method modes in the system.|
+|policies|[authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) collection|A collection of [authentication strength policies](authenticationStrengthPolicy.md) that exist for this tenant, including both built-in and custom policies.|
+
+## JSON representation
+The following is a JSON representation of the resource.
+<!-- {
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationStrengthRoot",
+  "baseType": "microsoft.graph.entity",
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthroot.md:41]: Updated

File: api-reference/v1.0/resources/authenticationstrengths-overview.md [hunk 23]
```diff
@@ -0,0 +1,84 @@
+---
+title: "Azure AD authentication strengths API overview"
+description: "Authentication strengths are sets of authentication method combinations that determine which methods your users can use to authenticate to access a resource."
+ms.localizationpriority: medium
+author: "mmcla"
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengths-overview.md:5]: Updated

File: api-reference/v1.0/resources/authenticationstrengths-overview.md [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+ms.prod: "identity-and-sign-in"
+doc_type: "conceptualPageType"
+ms.date: 12/10/2022
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengths-overview.md:8]: Updated to today, but this is likely going to be published in April. Is that OK?

File: api-reference/v1.0/resources/authenticationstrengthusage.md (hunk 3 above)
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthusage.md:1]: We need to add to the v1 PR, will keep this comment open until we address in the PR.

**Comment by InbarckMS** [comment]
> @FaithOmbongi - thanks for the review. I've gone over the comment and global comment and addressed the feedback.

Here is additional info regarding your question:
Question: Are updateAllowedCombinations action and findByMethodMode function getting deprecated in beta or is it that they won't be promoted to v1.0?
Answer: We ended up only deprecating findByMethodMode; I've asked the engineers to update the PR. Will circle back once it is done,

Add the authenticationStrengthPolicies relationship to policyRoot >> Done.

GLOBAL: The enums haven't been defined in the enums.md file  >> Done.

GLOBAL: RBAC --> I've responded to your comment with the permissions. Can you share an example I should follow?

GLOBAL: Delete all the # [HTTP](#tab/http) in examples. >> Done.

Couldn't find neither the usage function nor the authenticationStrengthUsage complex type in the v1.0 schema PR
+ Couldn't find the authenticationStrength relationship of conditionalAccessGrantControls in the schema either. 
 >> These are missing from the PR, waiting for the engineers to update and will circle back.

Please verify.
GLOBAL: Author - Update the docs to my name.

GLOBAL: None of the resources inherited from entity in v1.0.  >> Done.

GLOBAL: Lefovers /identity/conditionalAccess/authenticationStrengths/  >> Done.

changelog >> Done.

Please review, I will also circle back once the missing portions from the schema PR are updated.

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Hi @InbarckMS - A few feedback items, mostly clean up; See inline +
- GLOBAL: Restore all snippet files to what they were. The snippets will be autogenerated because the tooling will detect that your examples have changed.
- GLOBAL: Confirm the URLs in all the v1.0 docs. I see URLs for beta endpoints in v1.0 docs.
- For the RBAC statement, I've pushed a commit to create the include file and add a sample to one doc. For the READ statement, check out authenticationstrengthroot-list-policies.md; For WRITE statement, check out authenticationstrengthroot-delete-policies.md.

File: api-reference/v1.0/api/authenticationcombinationconfiguration-get.md [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+
+If successful, this method returns a `200 OK` response code and an [authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) object in the response body.
+
+## Examples
+
+### Request
+The following is an example of a request.
+
+``` http
+GET https://graph.microsoft.com/beta/identity/conditionalAccess/authenticationStrength/policies/0e371351-6419-4c8a-8047-61eef0212ffb/combinationConfigurations/133b68c4-503b-4e87-839a-6c286a27381b
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationcombinationconfiguration-get.md:57]: Updated.

File: api-reference/v1.0/api/authenticationcombinationconfiguration-update.md [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+
+|Property|Type|Description|
+|:---|:---|:---|
+|appliesToCombinations|authenticationMethodModes collection|The combinations to which this configuration applies. The only possible value for fido2combinationConfigurations is `fido2`. Required.|
+
+
+
+## Response
+
+If successful, this method returns a `204 NO CONTENT` response code.
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationcombinationconfiguration-update.md:54]: Updated.

File: api-reference/v1.0/api/authenticationcombinationconfiguration-update.md [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+
+## Examples
+
+### Request
+The following is an example of a request.
+
+``` http
+PATCH https://graph.microsoft.com/beta/identity/conditionalAccess/authenticationStrength/policies/0e371351-6419-4c8a-8047-61eef0212ffb/combinationConfigurations/4643f174-fe85-42b8-8b84-516775750a30
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationcombinationconfiguration-update.md:62]: Updated.

File: api-reference/v1.0/api/authenticationstrengthpolicy-list-combinationconfigurations.md [hunk 28]
```diff
@@ -0,0 +1,87 @@
+---
+title: "List combinationConfigurations"
+description: "Get the authenticationCombinationConfiguration objects for an authentication strength policy."
+author: "mmcla;inbarckMS"
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthpolicy-list-combinationconfigurations.md:4]: Updated.

File: api-reference/v1.0/api/authenticationstrengthpolicy-post-combinationconfigurations.md [hunk 29]
```diff
@@ -0,0 +1,100 @@
+---
+title: "Create authenticationCombinationConfiguration"
+description: "Create a new authenticationCombinationConfiguration object. In use, only fido2combinationConfigurations may be created, and these may only be created for custom authentication strength policies."
+author: "mmcla;inbarckMS"
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthpolicy-post-combinationconfigurations.md:4]: Updated.

File: api-reference/v1.0/api/authenticationstrengthpolicy-update.md [hunk 30]
```diff
@@ -0,0 +1,78 @@
...
+[!INCLUDE [table-intro](../../includes/update-property-table-intro.md)]
+
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String|The display name of the policy to be created. Optional.|
+|description|String|The description of the policy to be created. Optional.|
+
+## Response
+
+If successful, this method returns a `204 NO CONTENT` response code.
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthpolicy-update.md:50]: Updated

File: api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md [hunk 31]
```diff
@@ -20,9 +19,17 @@ One of the following permissions is required to call this API. To learn more, in
...
+
+> [!NOTE]
+> The `findByMethodMode` function is only available in `/beta` and will not be available in `/v1.0`. Use the following syntax instead.
+>
+> `GET /policies/authenticationStrengthPolicies?$filter[allowedCombinations/any(x:x has 'sms, password')]`Get a list of the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) objects and their properties filtered to only include policies that include the authentication method mode specified in the request.
 
+> [!NOTE]
+> The `findByMethodMode` function is only available in `/beta` and will not be available in `/v1.0`. Use the following syntax instead.
+>
+> `GET /policies/authenticationStrengthPolicies?$filter[allowedCombinations/any(x:x has 'sms, password')]`
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:32]: Updated

File: api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md [hunk 32]
```diff
@@ -20,9 +19,17 @@ One of the following permissions is required to call this API. To learn more, in
 |Permission type|Permissions (from least to most privileged)|
 |:---|:---|
 |Delegated (work or school account)|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
-|Delegated (personal Microsoft account)|Not supported.|
-|Application|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+Get a list of the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) objects and their properties filtered to only include policies that include the authentication method mode specified in the request.
```
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:22]: Updated

File: api-reference/v1.0/api/authenticationstrengthroot-list-policies.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/api/authenticationstrengthroot-list-policies.md:23]: RBAC is updated in beta and v1. Thanks for creating the includes files.

File: api-reference/beta/api/authenticationmethodmodedetail-get.md
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/beta/api/authenticationmethodmodedetail-get.md:1]: Updated

File: api-reference/beta/api/authenticationstrengthpolicy-list-combinationconfigurations.md
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/beta/api/authenticationstrengthpolicy-list-combinationconfigurations.md:1]: We have end point for /combinationConfigurations (for example FIDO2 advanced settings) and for /combinations  (that replaces authenticationCombinations). This is combinationConfigurations

File: api-reference/v1.0/resources/authenticationstrengthpolicy.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (InbarckMS) [reply api-reference/v1.0/resources/authenticationstrengthpolicy.md:24]: authenticationStrengthUsage is now added. Closing this comment.

File: api-reference/v1.0/resources/conditionalaccessgrantcontrols.md (hunk 12 above)
Comment (InbarckMS) [reply api-reference/v1.0/resources/conditionalaccessgrantcontrols.md:38]: Auth strength is now updated in the CA API schema. Closing this comment.

File: api-reference/beta/api/authenticationstrengthpolicy-list-combinationconfigurations.md
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/api/authenticationstrengthpolicy-list-combinationconfigurations.md:1]: Ah understood. Thanks.

**Comment by FaithOmbongi** [comment]
Hi @InbarckMS - Please use the latest changelog stubs. Because the engineer added a few more changes to his schema PR, there are some missing objects that should have been generated - including the deprecation of findByMethodMode for beta.

**Review by FaithOmbongi — APPROVED** [review]
Hi @InbarckMS - This is good to go when the schema is live.

I've retained the changes to the snippets. The team has confirmed that they'll overwrite and replace all of them but changing it in this PR has no effect.

**Comment by FaithOmbongi** [comment]
PR is on hold as we wait for the API to be deployed.

**Pull request merged**